packages/deploy/src/stateTransform.ts

```diff
@@ -20,7 +21,25 @@ import {
 import { DeployError } from './deployError';
 import { Logger } from '@openfn/logger/dist';
 
+function getStateJobCredential(
+  specJobCredential: undefined | null | string,
+  stateCredentials: ProjectState['project_credentials']
+): undefined | null | string {
+  if (
+    specJobCredential &&
+    typeof stateCredentials[specJobCredential] === 'undefined'
+  ) {
+    throw new DeployError(
+      `Could not find a credential with name: ${specJobCredential}`,
+      'VALIDATION_ERROR'
+    );
+  }
+
+  return specJobCredential && stateCredentials[specJobCredential].id;
```

Comment:
You've already tested that `specJobCredential` is defined - you can just go ahead and return the value here

Reply:
But it can be `null`